js/application.js

```diff
@@ -126,194 +209,507 @@ $(document).ready(function(){
+            if (prevElement.text()=='0') {
+              sum=Number(prevElement.text())+Number(currentElement.text());
+              prevElement.text(sum);
+              currentElement.text(0);
+              currentElement.attr('myAttr','canMove');
+              currentElement.attr('combine','canCombine');
+              prevElement.attr('combine','canCombine');
               var hasMoved=true;
-            } else if (($('#'+gridRowPrev).children('#'+gridCol).text()==$('#'+gridRow).children('#'+gridCol).text())&&($('#'+gridRowPrev).children('#'+gridCol).attr('combine')=='canCombine')&&($('#'+gridRow).children('#'+gridCol).attr('combine')=='canCombine')) {
-              sum=Number($('#'+gridRowPrev).children('#'+gridCol).text())+Number($('#'+gridRow).children('#'+gridCol).text());
-              $('#'+gridRowPrev).children('#'+gridCol).text(sum);
-              $('#'+gridRow).children('#'+gridCol).text(0);
+            } else if ((currentElement.text()==prevElement.text())&&(prevElement.attr('combine')=='canCombine')&&(currentElement.attr('combine')=='canCombine')) {
+              sum=Number(prevElement.text())+Number(currentElement.text());
+              prevElement.text(sum);
               totalScore += sum;
-              $('#'+gridRow).children('#'+gridCol).attr('myAttr','canMove');
-              $('#'+gridRow).children('#'+gridCol).attr('combine','canCombine');
-              $('#'+gridRowPrev).children('#'+gridCol).attr('combine','cantCombine');
+              currentElement.text(0);
+              currentElement.attr('myAttr','canMove');
+              currentElement.attr('combine','canCombine');
+              prevElement.attr('combine','cantCombine');
               var hasMoved=true;
             } else {
-              $('#'+gridRow).children('#'+gridCol).attr('myAttr','cantMove');
+              currentElement.attr('myAttr','cantMove');
             }
           } else {
-            $('#'+gridRow).children('#'+gridCol).attr('myAttr','cantMove');
+            currentElement.attr('myAttr','cantMove');
           }
         }
       }
     }
     reset();
-    if ((hasMoved==true)&&(winState==null)){randomNumberGeneration()};
+    if ((hasMoved==true)&&((winState==null)||(over==null))) {
+      randomNumberGeneration()
+    };
     printScore();
   }
 
   var checkWin = function(){
-    for(i=1; i<=4; i++){
-      var GridRow="grid_"+i;
-      for (j=1; j<=4; j++){
-        var GridCol="column"+j;
-        if ($('#'+GridRow).children('#'+GridCol).text()=='2048'){
-            winState="win";
-            console.log("You win!")
-        }     
+    if (max<=5){
+      for(i=1; i<=max; i++){
+        var GridRow="grid_"+i;
+        for (j=1; j<=max; j++){
+          var GridCol="column"+j;
+          if ($('#'+GridRow).children('#'+GridCol).text()==toWin){
+              winState="win";
+              $(".message").append("<h1>You win!</h1>");
+          }     
+        }
       }
     }
   }
 
   var checkLose = function(){
```

Comment:
A easier way to do this is to simulate `up`, `down`, `left`, `right` movement, and if it doesn't work, then you know that it's the end of the game.